dll/shellext/deskadp/deskadp.c

```diff
@@ -308,28 +309,40 @@ InitDisplayAdapterDialog(PDESKDISPLAYADAPTER This)
         LocalFree((HLOCAL)lpAdapterName);
     }
 
+    LoadString(hInstance,
+               IDS_NOTAVAIL,
+               lpNA,
+               _countof(lpNA));
+
     if (This->DeskExtInterface != NULL)
     {
         SetDlgItemTextW(This->hwndDlg,
                         IDC_CHIPTYPE,
-                        This->DeskExtInterface->ChipType);
+                        wcslen(This->DeskExtInterface->ChipType) ? This->DeskExtInterface->ChipType : lpNA);
         SetDlgItemTextW(This->hwndDlg,
                         IDC_DACTYPE,
-                        This->DeskExtInterface->DacType);
+                        wcslen(This->DeskExtInterface->DacType) ? This->DeskExtInterface->ChipType : lpNA);
```

Comment:
```suggestion
                        *(This->DeskExtInterface->DacType) ? This->DeskExtInterface->DacType : lpNA);
```

Reply:
Since the wcslen was just to verify that the string is "not empty", it's more efficient to just check whether the first character is not the null-terminator.
Also, there were copy-paste errors (the incorrect strings were being displayed).